### ISSUE TITLE
SVG symbols not in <defs>, 'url(#some-gradient)' is not found.

### ISSUE THREAD

**Issue by divStar** [g-makarov/vite-plugin-svg-spritemap#7]
Nice vite-plugin! However, I am having some troubles with it (and SVG in general, but that's a whole other topic).

In the automatically generated SVG spritemap the gradient is not displayed, not even the default values (in Firefox v120.0.1, MacOS 14). However: moving all symbols to `defs` element within the SVG spritemap (and moving all child `defs` elements to the top-level `defs`) makes it work. Therefore the manually edited SVG does display the default gradient values.

The other thing is, that I am not able to access either of these two in a way, that would allow me to e.g. adjust the colors using the embedded CSS3 variables and possibly start the animation (on the `#spinner`). I am currently a bit puzzled as to get it working, but I think this is not an issue with `vite-plugin-svg-spritemap`.

Would it make sense to move all `defs` contents to the top-level `defs` (including the `symbol` elements themselves)? It does change the "meaning" of the definitions though. If so, unless you have some time to spare, I could try to offer a PR.

<details>
<summary>Generated <code>icons.svg</code></summary>

```svg
<svg xmlns="http://www.w3.org/2000/svg">
    <symbol id="copy" viewBox="0 0 330 330">
        <g fill="var(--copy__fill)">
            <path d="M35 270h45v45c0 8.284 6.716 15 15 15h200c8.284 0 15-6.716 15-15V75c0-8.284-6.716-15-15-15h-45V15c0-8.284-6.716-15-15-15H35c-8.284 0-15 6.716-15 15v240c0 8.284 6.716 15 15 15m245 30H110V90h170zM50 30h170v30H95c-8.284 0-15 6.716-15 15v165H50z"></path>
            <path d="M155 120c-8.284 0-15 6.716-15 15s6.716 15 15 15h80c8.284 0 15-6.716 15-15s-6.716-15-15-15zM235 180h-80c-8.284 0-15 6.716-15 15s6.716 15 15 15h80c8.284 0 15-6.716 15-15s-6.716-15-15-15M235 240h-80c-8.284 0-15 6.716-15 15 0 8.284 6.716 15 15 15h80c8.284 0 15-6.716 15-15 0-8.284-6.716-15-15-15"></path>
        </g>
    </symbol>
    <symbol id="spinner">
        <defs>
            <linearGradient id="a" x1="8.042%" x2="65.682%" y1="0%" y2="23.865%">
                <stop offset="0%" stop-color="var(--spinner__stop0, red)" stop-opacity="0"></stop>
                <stop offset="63.146%" stop-color="var(--spinner__stop1, red)" stop-opacity=".631"></stop>
                <stop offset="100%" stop-color="var(--spinner__stop2, red)"></stop>
            </linearGradient>
        </defs>
        <g fill="none" fill-rule="evenodd" transform="translate(1 1)">
            <path stroke="url(#a)" stroke-width="2" d="M36 18c0-9.94-8.06-18-18-18">
                <animateTransform attributeName="transform" dur="0.9s" from="0 18 18" repeatCount="indefinite"
                                  to="360 18 18" type="rotate"></animateTransform>
            </path>
            <circle cx="36" cy="18" r="1" fill="var(--spinner__leading-dot, white)">
                <animateTransform attributeName="transform" dur="0.9s" from="0 18 18" repeatCount="indefinite"
                                  to="360 18 18" type="rotate"></animateTransform>
            </circle>
        </g>
    </symbol>
</svg>
```
</details>

<details>
<summary>Manually edited <code>icons.svg</code></summary>

```svg
<svg xmlns="http://www.w3.org/2000/svg">
    <defs>
        <linearGradient id="a" x1="8.042%" x2="65.682%" y1="0%" y2="23.865%">
            <stop offset="0%" stop-color="var(--spinner__stop0, red)" stop-opacity="0"></stop>
            <stop offset="63.146%" stop-color="var(--spinner__stop1, red)" stop-opacity=".631"></stop>
            <stop offset="100%" stop-color="var(--spinner__stop2, red)"></stop>
        </linearGradient>
        <symbol id="copy" viewBox="0 0 330 330">
            <g fill="var(--copy__fill)">
                <path d="M35 270h45v45c0 8.284 6.716 15 15 15h200c8.284 0 15-6.716 15-15V75c0-8.284-6.716-15-15-15h-45V15c0-8.284-6.716-15-15-15H35c-8.284 0-15 6.716-15 15v240c0 8.284 6.716 15 15 15m245 30H110V90h170zM50 30h170v30H95c-8.284 0-15 6.716-15 15v165H50z"></path>
                <path d="M155 120c-8.284 0-15 6.716-15 15s6.716 15 15 15h80c8.284 0 15-6.716 15-15s-6.716-15-15-15zM235 180h-80c-8.284 0-15 6.716-15 15s6.716 15 15 15h80c8.284 0 15-6.716 15-15s-6.716-15-15-15M235 240h-80c-8.284 0-15 6.716-15 15 0 8.284 6.716 15 15 15h80c8.284 0 15-6.716 15-15 0-8.284-6.716-15-15-15"></path>
            </g>
        </symbol>
        <symbol id="spinner">
            <g fill="none" fill-rule="evenodd" transform="translate(1 1)">
                <path stroke="url(#a)" stroke-width="2" d="M36 18c0-9.94-8.06-18-18-18">
                    <animateTransform attributeName="transform" dur="0.9s" from="0 18 18" repeatCount="indefinite"
                                      to="360 18 18" type="rotate"></animateTransform>
                </path>
                <circle cx="36" cy="18" r="1" fill="var(--spinner__leading-dot, white)">
                    <animateTransform attributeName="transform" dur="0.9s" from="0 18 18" repeatCount="indefinite"
                                      to="360 18 18" type="rotate"></animateTransform>
                </circle>
            </g>
        </symbol>
    </defs>
</svg>
```
</details>

**Comment by g-makarov** [comment]
Hi, thanks for the feedback :)
I think it's a good idea to move `<defs />` to the top level. I've already made a new version 0.0.13 with it, please check if it works well. But now it will only work with the `currentColor: false` parameter in the plugin definition

**Comment by g-makarov** [comment]
released in version [0.0.13](https://github.com/g-makarov/vite-plugin-svg-spritemap/commit/9ee8f0d56db45864edaa569a872e5bbdf7d0306a)
